### PR TITLE
Add definitions for 'test_data', 'training_data' (English)

### PR DESCRIPTION
**Pull request by MikeResearch**
added test_data and training_data definitions

Fill in each of the sections (using NA for those that are not applicable).

## Author: 
Mike Ramshaw
- 

## Language: 
English
- 

## Terms defined:
test_data, training_data
- 
- 
-

### CODE REVIEW

File: glossary.yml [hunk 1]
```diff
@@ -5892,7 +5892,15 @@
...
-
+  ref:
+    - machine_learning
+  en:
+    term: "test data"
+    def: >
+      Test data is a portion of a dataset used to evaluate the correctness of a 
+      [machine learning](#machine_learning) algorithm after it has been trained. It should always
+      be separated from the [training data](#training_data) to ensure that the model is 
+      properly tested with unseen data.
```
Comment (baileythegreen) [inline glossary.yml:5903]:
```suggestion
      [machine learning](#machine_learning) algorithm after it has been trained.
      It should always be separated from the [training data](#training_data) to
      ensure that the model is properly tested with unseen data.
```

Changing line width.

File: glossary.yml [hunk 2]
```diff
@@ -6028,7 +6036,15 @@
...
-
+  ref:
+    - machine_learning
+  en:
+    term: "training data"
+    def: >
+      Training data is a portion of a dataset used to train a
+      [machine learning](#machine_learning) algorithm to recognize similar data. It should always
+      be separated from the [test data](#test_data) to ensure that the model is 
+      properly tested with data it has never seen before.
```
Comment (baileythegreen) [inline glossary.yml:6047]:
```suggestion
      Training data is a portion of a dataset used to train [machine learning](#machine_learning)
      algorithm to recognise similar data. It should always be separated from the
      [test data](#test_data) to ensure that the model is properly tested with data it has
      never seen before.
```

Changing line widths and American to British spelling.

**Review by baileythegreen — APPROVED** [review]

**Comment by baileythegreen** [comment]
Thanks! If you contribute more in future, know that we prefer British spellings in this project, and that, while there is no set width for lines, we do try to keep them approximately the same.

**Comment by elletjies** [comment]
@all-contributors please add @MikeResearch for translation

**Pull request merged**